app/src/test/java/com/amaze/filemanager/filesystem/usb/ReflectionHelpers.java

```diff
@@ -0,0 +1,174 @@
+    static UsbDevice callUsbDeviceConstructor(@NonNull String name, int vendorId, int productId,
+                                               int usbClass, int subClass, int protocol,
+                                               @Nullable String manufacturerName,
+                                               @Nullable String productName,
+                                               @NonNull String version, @Nullable String serialNumber)
+            throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
+
+        Class<UsbDevice> clazz = (Class<UsbDevice>) Class.forName("android.hardware.usb.UsbDevice");
+        Constructor<UsbDevice> constructor = clazz.getConstructor(String.class,
+                int.class, int.class, int.class, int.class, int.class, String.class, String.class,
+                String.class, String.class);
+
+        return constructor.newInstance(name, vendorId, productId, usbClass, subClass, protocol,
+                manufacturerName, productName, version, serialNumber);
+    }
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    static UsbDevice callUsbDeviceConstructor(@NonNull String name, int vendorId, int productId,
+                                              int usbClass, int subClass, int protocol,
+                                              @Nullable String manufacturerName,
+                                              @Nullable String productName, @Nullable String serialNumber)
+            throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
+
+        Class<UsbDevice> clazz = (Class<UsbDevice>) Class.forName("android.hardware.usb.UsbDevice");
+        Constructor<UsbDevice> constructor = clazz.getConstructor(String.class,
+                int.class, int.class, int.class, int.class, int.class, String.class, String.class,
+                String.class);
+
+        return constructor.newInstance(name, vendorId, productId, usbClass, subClass, protocol,
+                manufacturerName, productName, serialNumber);
+    }
+
+    @RequiresApi(Build.VERSION_CODES.KITKAT)
+    static UsbDevice callUsbDeviceConstructor(@NonNull String name, int vendorId, int productId,
+                                              int usbClass, int subClass, int protocol,
+                                              @NonNull Parcelable[] interfaces)
+            throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
+
+        Class<UsbDevice> clazz = (Class<UsbDevice>) Class.forName("android.hardware.usb.UsbDevice");
+        Constructor<UsbDevice> constructor = clazz.getConstructor(String.class,
+                int.class, int.class, int.class, int.class, int.class, Parcelable[].class);
+
+        return constructor.newInstance(name, vendorId, productId, usbClass, subClass, protocol);
```

Comment:
`interfaces` was omitted in calling the constructor, causing the test to fail at API 19 (not the exception you got though)

Reply:
Fixed!